File: tests/basic/instr_accumulate.ml
open Optitrust
open Target


let _ = Run.doc_script_cpp (fun _ ->
    !! Instr_basic.accumulate [cLabel "fuse"];
  )
"
int main() {
  int x = 0;
  fuse: {
    x += 1;
    x += 2;
    x += 3;
  }
}
"

let _ = Run.script_cpp (fun _ ->

  !! Instr_basic.accumulate [cLabel "test1"];
  !! Instr_basic.accumulate [cLabel "test2"];

)

File: tests/basic/instr_accumulate.cpp

typedef struct {
  int x;
  int y;

} vect;


int main () {

  int x = 10;
  test1: {
    x = x + 2;
    x = x + 3;
    x = x + 4;
    x = x + 5;
    x = x + 6;
    x += 7;
  }
  vect result = {0,0};
  test2: {
    result.x += 2;
    result.x += 3;
    result.x += 4;
    result.x += 5;
    result.x += 6;
    result.x += 7;
  }
  return 0;
}